spring-kafka/src/main/java/org/springframework/kafka/listener/TimestampedException.java

```diff
@@ -34,14 +34,41 @@
 
 	private final long timestamp;
 
+	@Deprecated
 	public TimestampedException(Exception ex, Clock clock) {
 		super(ex.getMessage(), ex);
 		this.timestamp = Instant.now(clock).toEpochMilli();
 	}
 
+	/**
+	 * Constructs an instance with the provided cause
+	 * and the current time.
+	 * @param ex the exception cause.
+	 */
 	public TimestampedException(Exception ex) {
-		super(ex.getMessage(), ex);
-		this.timestamp = Instant.now(Clock.systemDefaultZone()).toEpochMilli();
+		this(ex, Instant.now());
+	}
+
+	/**
+	 * Creates an instance with the timestamp of when it was thrown and its cause.
+	 * @param ex the exception cause.
+	 * @param timestamp the millis from epoch of when the exception was thrown.
+	 * @since 2.8.5
```

Comment:
So, back-port to `2.7.x` label on the issue is wrong. Isn't it?

Reply:
Well, not sure really, I hadn't noticed that. If it's meant to be back-ported, what should I change this to? Maybe 2.7.x would be confusing since 2.8.0 - 2.8.4 doesn't have that? Not that proficient with some of these criteria yet.

Reply:
I guess this one `@since 2.7.13` as the current point release for that branch.
There is no reason to mention since what `2.8.x` or `2.9.x` just because you won't see this API in the `2.8.0` anyway.
Therefore if you see it like `@since 2.7.13` in your code base that means it is included into the release of `2.8.x` or `2.9.x`.
It might even not give any gain for end-users, unless there is a bug and it will be much easier to report the bug for the API which is marked with respective `@since`.
We then on our side will figure out what branches the fix belongs too.